### PR TITLE
PullSubscription Status reflect non-creatable Deployment

### LINKED ISSUE #1682: PullSubscription Status does not reflect non-creatable Deployment
**Describe the bug**
When PullSubscriptions cannot create their Deployment, the PullSubscription's status does not have a reasonable error message.

 **Expected behavior**
The PullSubscription's status should have something like "Unable to create/update Deployment: [error when creating Deployment]"

 **To Reproduce**
Create a PullSubscription that whose Deployment cannot be created.

```shell
gcloud pubsub topics create my-topic
gcloud beta events brokers create default

cat << END | kubectl apply -f -
apiVersion: internal.events.cloud.google.com/v1beta1
kind: PullSubscription
metadata:
  name: ps
spec:
  ackDeadline: 30s
  adapterType: pubsub
  mode: CloudEventsBinary
  secret:
    key: "" # This is the actual error that prevents the Deployment from being created.
  serviceAccountName: default
  sink:
    ref:
      apiVersion: eventing.knative.dev/v1beta1
      kind: Broker
      name: default
  topic: my-topic
END
```

```
kubectl describe pullsubscription ps
```

 **Additional context**
```
apiVersion: internal.events.cloud.google.com/v1beta1
kind: PullSubscription
metadata:
  annotations:
    metrics-resource-group: cloudpubsubsources.events.cloud.google.com
  creationTimestamp: "2020-09-10T16:50:01Z"
  finalizers:
  - pullsubscriptions.internal.events.cloud.google.com
  generation: 1
  labels:
    events.cloud.google.com/source-name: source-for-my-t2
    receive-adapter: cloudpubsubsource.events.cloud.google.com
  name: source-for-my-t2
  namespace: default
  ownerReferences:
  - apiVersion: events.cloud.google.com/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: CloudPubSubSource
    name: source-for-my-t2
    uid: b982e8ae-916c-4530-9003-378f377160ce
  resourceVersion: "594193"
  selfLink: /apis/internal.events.cloud.google.com/v1beta1/namespaces/default/pullsubscriptions/source-for-my-t2
  uid: 2883d926-a41c-4b01-b532-fbaa4eac8fb1
spec:
  ackDeadline: 30s
  adapterType: pubsub
  ceOverrides:
    extensions:
      knsourcetrigger: link0.4172274926105507
  mode: CloudEventsBinary
  retentionDuration: 168h0m0s
  secret:
    key: "" # This is the actual error that prevents the Deployment from being created.
  serviceAccountName: foobar
  sink:
    ref:
      apiVersion: eventing.knative.dev/v1beta1
      kind: Broker
      name: default
  topic: my-topic2
status:
  conditions:
  - lastTransitionTime: "2020-09-10T16:50:03Z"
    status: Unknown
    type: Deployed
  - lastTransitionTime: "2020-09-10T16:50:03Z"
    status: Unknown
    type: Ready
  - lastTransitionTime: "2020-09-10T16:50:03Z"
    status: "True"
    type: SinkProvided
  - lastTransitionTime: "2020-09-10T16:50:03Z"
    status: "True"
    type: Subscribed
  - lastTransitionTime: "2020-09-10T16:50:03Z"
    message: Transformer is nil
    reason: TransformerNil
    severity: Info
    status: "False"
    type: TransformerProvided
  observedGeneration: 1
  projectId: harwayne2
  sinkUri: http://default-brokercell-ingress.cloud-run-events.svc.cluster.local/default/default
  subscriptionId: cre-src_default_source-for-my-t2_2883d926-a41c-4b01-b532-fbaa4eac8fb1
```
I think we just need to write the status here, in addition to a k8s event. 
https://github.com/google/knative-gcp/blob/038c81ca09793d2000e0016a304a2951594b82d9/pkg/reconciler/intevents/pullsubscription/reconciler.go#L150

We likely also need to mutate the error that is passed here to say whether it was from trying to create or update the Deployment.

### PR DESCRIPTION
**Pull request by capri-xiyue**
Fixes #1682 

## Proposed Changes

- PullSubscription Status reflect non-creatable Deployment
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by capri-xiyue** [comment]
Screen Shot 2020-09-16 at 5 45 51 PM

File: pkg/reconciler/intevents/pullsubscription/reconciler.go [hunk 1]
```diff
@@ -328,11 +328,13 @@ func (r *Base) GetOrCreateReceiveAdapter(ctx context.Context, desired *appsv1.De
 	existing, err := r.getReceiveAdapter(ctx, ps)
 	if err != nil && !apierrors.IsNotFound(err) {
 		logging.FromContext(ctx).Desugar().Error("Unable to get an existing Receive Adapter", zap.Error(err))
+		ps.Status.MarkDeployedUnknown("Error getting an existing Receive Adapter", err.Error())
 		return nil, err
 	}
 	if existing == nil {
 		existing, err = r.KubeClientSet.AppsV1().Deployments(ps.Namespace).Create(desired)
 		if err != nil {
+			ps.Status.MarkDeployedFailed("Error creating an Receive Adapter", err.Error())
```
Comment (nachocano) [inline pkg/reconciler/intevents/pullsubscription/reconciler.go:337]: nit: "Error create a Receive Adapter"

Comment (nachocano) [reply pkg/reconciler/intevents/pullsubscription/reconciler.go:337]: actually, change it afterwards with the UTs

File: pkg/reconciler/intevents/pullsubscription/keda/pullsubscription.go [hunk 2]
```diff
@@ -81,6 +81,7 @@ func (r *Reconciler) ReconcileScaledObject(ctx context.Context, ra *appsv1.Deplo
 		existing.Spec = ra.Spec
 		existing, err = r.KubeClientSet.AppsV1().Deployments(src.Namespace).Update(existing)
 		if err != nil {
+			src.Status.MarkDeployedFailed("Error updating the Receive Adapter", err.Error())
```
Comment (Harwayne) [inline pkg/reconciler/intevents/pullsubscription/keda/pullsubscription.go:84]: Shouldn't this be 

```
src.Status.MarkDeployedFailed("ReceiveAdapterUpdateFailed", "Error updating the Receive Adapter: %s", err.Error())
```
?

As is, it is missing the reason and the format string doesn't include any format directives.

Comment (capri-xiyue) [reply pkg/reconciler/intevents/pullsubscription/keda/pullsubscription.go:84]: Thanks for catching this. Fixed with this PR https://github.com/google/knative-gcp/pull/1721

**Pull request merged**